projects/kit/src/lib/utils/date/tests/append-date.spec.ts

```diff
@@ -0,0 +1,181 @@
+
+        expect(result.getFullYear()).toBe(2000);
+        expect(result.getMonth()).toBe(6);
+        expect(result.getDate()).toBe(15);
+    });
+
+    it('year: 2005, month: 6, day: 14, if {year: 5} was passed', () => {
+        const result = appendDate(y2000m6d15, {year: 5});
+
+        expect(result.getFullYear()).toBe(2005);
+        expect(result.getMonth()).toBe(6);
+        expect(result.getDate()).toBe(14);
+    });
+
+    it('year: 1995, month: 6, day: 14, if {year: -5} was passed', () => {
+        const result = appendDate(y2000m6d15, {year: -5});
+
+        expect(result.getFullYear()).toBe(1995);
+        expect(result.getMonth()).toBe(6);
+        expect(result.getDate()).toBe(14);
+    });
+
+    it('year: 2000, month: 11, day: 14, if {month: 5} was passed', () => {
+        const result = appendDate(y2000m6d15, {month: 5});
+
+        expect(result.getFullYear()).toBe(2000);
+        expect(result.getMonth()).toBe(11);
+        expect(result.getDate()).toBe(14);
+    });
+
+    it('year: 2000, month: 1, day: 14, if {month: -5} was passed', () => {
+        const result = appendDate(y2000m6d15, {month: -5});
+
+        expect(result.getFullYear()).toBe(2000);
+        expect(result.getMonth()).toBe(1);
+        expect(result.getDate()).toBe(14);
+    });
+
+    it('year: 2000, month: 6, day: 19, if {day: 5} was passed', () => {
+        const result = appendDate(y2000m6d15, {day: 5});
+
+        expect(result.getFullYear()).toBe(2000);
+        expect(result.getMonth()).toBe(6);
+        expect(result.getDate()).toBe(19);
+    });
+
+    it('year: 2000, month: 6, day: 9, if {day: -5} was passed', () => {
+        const result = appendDate(y2000m6d15, {day: -5});
+
+        expect(result.getFullYear()).toBe(2000);
+        expect(result.getMonth()).toBe(6);
+        expect(result.getDate()).toBe(9);
+    });
+
+    it('year: 2000, month: 6, day: 31, if {day: 17} was passed', () => {
+        const result = appendDate(y2000m6d15, {day: 17});
+
+        expect(result.getFullYear()).toBe(2000);
+        expect(result.getMonth()).toBe(6);
+        expect(result.getDate()).toBe(31);
+    });
+
+    it('year: 2000, month: 11, day: 30, if {day: 169} was passed', () => {
+        const result = appendDate(y2000m6d15, {day: 169});
+
+        expect(result.getFullYear()).toBe(2000);
+        expect(result.getMonth()).toBe(11);
+        expect(result.getDate()).toBe(30);
+    });
+
+    it('year: 2000, month: 11, day: 31, if {day: 170} was passed', () => {
+        const result = appendDate(y2000m6d15, {day: 170});
+
+        expect(result.getFullYear()).toBe(2000);
+        expect(result.getMonth()).toBe(11);
+        expect(result.getDate()).toBe(31);
+    });
+
+    it('year: 1999, month: 11, day: 30, if {day: -197} was passed', () => {
+        const result = appendDate(y2000m6d15, {day: -197});
+
+        expect(result.getFullYear()).toBe(1999);
+        expect(result.getMonth()).toBe(11);
+        expect(result.getDate()).toBe(30);
```

Comment:
Should it be `01.01.2000` ?
<img width="1312" height="603" alt="day-197" src="https://github.com/user-attachments/assets/53baf44d-d7f1-4f8f-bd68-fc04545ccaee" />

https://stackblitz.com/edit/tui-calendar-range-min-length-d197?file=src%2Fapp%2Fapp.component.ts